torch/optim/optimizer.py

```diff
@@ -54,6 +54,15 @@ def __repr__(self) -> str:
 required = _RequiredParameter()
 
 
+def _disable_dynamo_if_closure(fn):
+    def inner(self, *args, **kwargs):
+        if (len(args) == 1 and args[0] is not None) or (len(kwargs) == 1 and kwargs.get("closure", None) is not None):
+            return torch._disable_dynamo(functools.partial(fn))(self, *args, **kwargs)
```

Comment:
How about simply

`return torch._disable_dynamo(fn)(self, *args, **kwargs)`

Reply:
that will disable the step forever, won't it?

Reply:
I don't get the last comment. If you use two optimizer instances, both will have their own decorators.

Reply:
won't step be disabled forever for that instance? I was under the impression that it mutated the function that is disabled (ie if recursive is False I see skip and stuff being set) if recursive=True does it not mutate the function instance?

Reply:
yeah you right, just tried it, doesn't look like it mutates, I'll remove